### ISSUE TITLE
Building pressure sensors

### ISSUE THREAD

**Issue by philrosenberg** [philrosenberg/instrument-log#35]
The pressure sensors will each be housed in a box for waterproofing. The boxes will each have two rj45 connectors to enable piggybacking via the rs485 interface. These connectors will carry both data and power.

I will need a logger. This could be a laptop, but a Moxa UC-5101-LX would also be a good option.

I will be using cat 6a cable I think as we need long lengths ~100m, so I want to minimise the impact of noise

**Comment by philrosenberg** [comment]
Here is a shopping list for each sensor
Fibox PCM 150/100 T box RS: 498-5129
Fibox mounting plate MNX series 148x98x1.5 RS: 504-192
2 of Phoenix waterproof rj45 female 1689064, 1689844 and 1652606 RS: 648-6674, 652-6895 and 648-4375
15 pin d-sub high density screw terminal connector clever little box CLB-VGA15F-16TB-2 CPC/Farnell: CN21719
9 pin d-sub screw terminal connector clever little box CLB-CSUB09F-10TB-2 CPC/Farnell: CN21716
a patch cable to cut in half and connect from the rj45 to the screw terminals

for the cables I uses these connectors
Phoenix 1656990 RS: 607-8318

The 9 pin d-sub means I can have a standard rs-232 connector inside the box for lab use as well as the rj45 for field use

**Comment by philrosenberg** [comment]
Just working out how to wire things. The Moxa UC-5101 uses the following pins on it's RJ45 connector
2 - TX+
3 - GND
4 - TX-
5 - RX+
6 - RX-

This is a bit weird as 2 and 4 aren't even the same twisted pair.

I'm going to use something more like the ethernet standard with PoE. seems a lot more sensible. Then I'll do something to convert to the moxa standard if needed.

**Comment by philrosenberg** [comment]
I set up the sensors on a rs485 network. This was a two wire network using ethernet cable. I used green as both Tx- and Rx-. I used green/white as both Tx+ and Rx+. I used both brown and brown/white as +ve power and blue and blue/white as -ve power. Orange and orange/white weren't used.

I added a 100 ohm resistor across the data +/- terminal of unit 4 to avoid reflections on the data lines as this is 100ohm characteristic inductance cable (check the definition of characteristic inductance if you don't know it - I didn't)

Basically this didn't work. If the sensors are put in continuous output mode, they tread all over each other and corrupt the data.

If I poll the sensors to fast then they quickly stop responding.

I'm going to need to run each on a separate rs485 port.

**Comment by philrosenberg** [comment]
Returning to this some time later
I Bought 4 separate rs485 to usb adaptors. They were FTDI Chip RS485 USB A Male to Cable End Converter Cable, part number USB-RS485-WE-1800-BT, RS part number 687-7834. I've attached the data sheet.

[0900766b80da71f7.pdf](https://github.com/user-attachments/files/19687908/0900766b80da71f7.pdf)

I connected two of these to one network cable and ran the signal for two microbarometers down one cable. The daisy chain ports on the microbarometers permitted this.

I made a small box to connect the microbarometer cables to the RS485 usb converters. Here is the wiring ued:
Red USB - power out - sleaved and not used.
Brown and green USB - placed a 100 ohm resistor across these. The manual said to use 120, but that is the value for serial cable - ethernet cable, which I used has a 100 ohm impedance, so I used that size resistor.
Orange USB - data+ - connected to white/green for one sensor or white/orange for the other
Yellow USB - data- - connected to green for one sensor and orange for the other

In addition, I had power going down the network cable. This was positive on BOTH brown and brown/white and negative on BOTH blue and blue/white. This halved the resistance of the cable and reduced voltage drop. I think it also matches with Power over Ethernet. Power was supplied by a 9V transformer.

**Comment by philrosenberg** [comment]
Note that I then also at the pressure sensor end connected the Tx and Rx together making this a two wire setup, rather than a 4 wire setup.
